### ISSUE TITLE
Log parser warnings do not point to the right file in multi-file project

### ISSUE THREAD

**Issue by teored90** [James-Yu/LaTeX-Workshop#223]
### Description

In a multi-file project with a main tex file and some auxiliary files included either with `\input` or `\include`, warnings in the Problems window do not point to the auxiliary file, but to the main file.

I guess this is a limitation in the parsing of the latex log file, complicated by the fact that for warnings, the name of the file is not reported on the same line as the warning message.

### How to reproduce

Create a `main.tex` file with content:

```
\documentclass{article}

\begin{document}
\title{Hello World}
\maketitle
 
This is a warning in the main file \ref{nonexisting_label_main}

\include{myfile}
\end{document}
```

and a `myfile.tex` file with content

    This is raises a warning in myfile.tex \ref{nonexisting_label}.

Then compile.

### Expected behavior
These two warnings should be raised
```
file: 'file:///Users/matteo/Desktop/vscodebug/myfile.tex'
severity: 'Warning'
message: 'Reference `nonexisting_label' on page 2 undefined '
at: '1,1'
source: 'LaTeX'
```

```
file: 'file:///Users/matteo/Desktop/vscodebug/main.tex'
severity: 'Warning'
message: 'Reference `nonexisting_label_main' on page 1 undefined '
at: '7,1'
source: 'LaTeX'
```

### Actual behavior
The first one points to `main.tex` instead of `myfile.tex`

```
file: 'file:///Users/matteo/Desktop/vscodebug/main.tex'
severity: 'Warning'
message: 'Reference `nonexisting_label' on page 2 undefined '
at: '1,1'
source: 'LaTeX'
```

```
file: 'file:///Users/matteo/Desktop/vscodebug/main.tex'
severity: 'Warning'
message: 'Reference `nonexisting_label_main' on page 1 undefined '
at: '7,1'
source: 'LaTeX'
```

### Log messages

There is nothing relevant in the log messages.

**Comment by James-Yu** [comment]
Here is the compiler log, which imo is extremely hard to parse:
```
Latexmk: applying rule 'pdflatex'...
This is pdfTeX, Version 3.14159265-2.6-1.40.17 (TeX Live 2016/W32TeX) (preloaded format=pdflatex)
 restricted \write18 enabled.
entering extended mode
(c:/Users/James/Desktop/test/exmp01.tex
LaTeX2e <2017/01/01> patch level 3
Babel <3.9r> and hyphenation patterns for 83 language(s) loaded.
(d:/TexLive/2016/texmf-dist/tex/latex/base/article.cls
Document Class: article 2014/09/29 v1.4h Standard LaTeX document class
(d:/TexLive/2016/texmf-dist/tex/latex/base/size10.clo)) (./exmp01.aux
(./myfile.aux))

LaTeX Warning: No \author given.


LaTeX Warning: Reference `nonexisting_label_main' on page 1 undefined on input line 7.

[1{d:/TexLive/2016/texmf-var/fonts/map/pdftex/updmap/pdftex.map}] (./myfile.tex

LaTeX Warning: Reference `nonexisting_label' on page 2 undefined on input line 1.

) [2] (./exmp01.aux (./myfile.aux))

LaTeX Warning: There were undefined references.

 )<d:/TexLive/2016/texmf-dist/fonts/type1/public/amsfonts/cm/cmbx10.pfb><d:/TexLive/2016/texmf-dist/fonts/type1/public/amsfonts/cm/cmr10.pfb><d:/TexLive/2016/texmf-dist/fonts/type1/public/amsfonts/cm/cmr12.pfb><d:/TexLive/2016/texmf-dist/fonts/type1/public/amsfonts/cm/cmr17.pfb>
Output written on exmp01.pdf (2 pages, 40871 bytes).
SyncTeX written on exmp01.synctex.gz.
Transcript written on exmp01.log.
=== TeX engine is 'pdfTeX'
Latexmk: All targets (exmp01.pdf) are up-to-date
```

It seems that
```
[1{d:/TexLive/2016/texmf-var/fonts/map/pdftex/updmap/pdftex.map}] (./myfile.tex

LaTeX Warning: Reference `nonexisting_label' on page 2 undefined on input line 1.

)
```
should be interpreted as
```
[1{something}] (./myfile.tex LaTeX Warning: Reference `nonexisting_label' on page 2 undefined on input line 1.)
```
However, the first warning is simply presented in the form of
```
LaTeX Warning: Reference `nonexisting_label_main' on page 1 undefined on input line 7.
```
which breaks the consistency.

Are there any other tex editor that can properly parse such log messages? We may borrow their idea to enhance our parser.

**Comment by teored90** [comment]
If you add another undefined reference in myfile.tex, the resulting log is
```
[...]
(./myfile.tex

LaTeX Warning: Reference `nonexisting_label' on page 2 undefined on input line 
1.


LaTeX Warning: Reference `another_nonexisting_label' on page 2 undefined on inp
ut line 2.

) 

[...]
```

So I think pdflatex groups the warnings between round brackets, with the filename right next to the opening bracket (excluding the warnings in the main file).

The plugin attom-latextools for Atom does a decent parsing of the warnings, but I don't understand the coffee syntax that much. Here is the log parser code: https://github.com/msiniscalchi/atom-latextools/blob/master/lib/parsers/parse-tex-log.coffee.

**Comment by teored90** [comment]
Also the TextMate (a OS X text editor) Latex bundle does parse the file correctly. Here is the Python code for the parser: https://github.com/textmate/latex.tmbundle/blob/master/Support/lib/Python/parsing.py

**Comment by lihuanglx** [comment]
This repo might be helpful: https://github.com/sharelatex/latex-log-parser-sharelatex
